### PR TITLE
feat: adapt window reader to order rules

### PR DESCRIPTION
**Pull request by v0y4g3r**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

This PR brigdges order-rules with windowed reader. Now we can infer scan windows by order options.

If The first column in order-by clause is timestamp, we can optimize it into a windowed table scan so that it can terminate as soon as we found enough records.

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [ ]  I have added the necessary unit tests and integration tests.

## Refer to a related PR or issue link (optional)
#1286

### CODE REVIEW

File: src/storage/src/region/tests.rs
Comment (waynexia) [inline src/storage/src/region/tests.rs:1]: It looks like sorting by ascending ordering is not covered

**Review by waynexia — APPROVED** [review]
LGTM

File: src/storage/src/read/windowed.rs [hunk 1]
```diff
@@ -133,13 +148,11 @@ fn sort_by_rows(schema: &ProjectedSchemaRef, arrays: Vec<ArrayRef>) -> Result<Ve
     Ok(sorted)
 }
 
-/// [<PK_1>, <PK_2>, TS] to [TS, <PK_1>, <PK_2>].
+/// Builds sort columns from `order_options`.
```
Comment (killme2008) [inline src/storage/src/read/windowed.rs:151]: sort? Looks like `sorted` is more suitable. Both in comment and function name.

**Review by killme2008 — APPROVED** [review]
LGTM

**Pull request merged**